version: '3.8'
services:

  frontend:
    build:
      context: ./frontend
    ports:
      - "5173:5173"
    env_file:
      - frontend/.env        
    environment: 
      - VITE_API_URL=http://ficteer.com
    networks:
      - mine 

  backend:
    build:
      context: ./backend
    ports:
      - "8000:8000"
    env_file:
      - backend/.env
    environment:
      PYTHONPATH: /app
    depends_on:
      - db
    networks:
      - mine

  db:
    image: postgres:13
    volumes:
      - ./init-db.sh:/docker-entrypoint-initdb.d/init-db.sh
    environment:
      POSTGRES_DB: app
      POSTGRES_USER: app
      POSTGRES_PASSWORD: changethis123
    ports:
      - "5432:5432"
    networks:
      - mine

  adminer:
    image: adminer
    ports:
      - "8081:8080"
    networks:
      - mine

  proxy:
    image: 'docker.io/jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      - '81:81'
      - '443:443'
    volumes:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt
    networks:
      - mine

  nginx:
    image: nginx:latest
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
      - ./proxy_params.conf:/etc/nginx/proxy_params.conf
    ports:
      - "80:80"
    depends_on:
      - frontend
      - backend
      - db
      - adminer
      - proxy
    networks:
      - mine 

networks:
  mine:
    driver: bridge
